sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample_MockClient.md

```diff
@@ -0,0 +1,88 @@
+# Mock a client for testing using the Moq library
+
+This sample illustrates how to use [Moq][moq] to create a unit test that mocks the response from a `FormRecognizerClient` method. For more examples of mocking, see [Moq samples][moq_samples].
+
+## Define a method that uses FormRecognizerClient
+To show the usage of mocks, define a method that will be tested with mocked objects. For this case, we are going to create a method that will verify whether the total price of a groceries list is expensive or not. We're not considering it expensive if the confidence of the recognized field is too low.
+
+```C# Snippet:FormRecognizerMethodToTest
+private static async Task<bool> IsExpensiveAsync(string modelId, Uri documentUri, FormRecognizerClient client)
+{
+    RecognizeCustomFormsOperation operation = await client.StartRecognizeCustomFormsFromUriAsync(modelId, documentUri);
+
+    Response<RecognizedFormCollection> response =  await operation.WaitForCompletionAsync();
+    RecognizedForm form = response.Value[0];
+
+    if (form.FormType == "groceries")
+    {
+        FormField totalPriceField = form.Fields["totalPrice"];
+        return totalPriceField.Confidence > 0.5f && totalPriceField.Value.AsFloat() > 100f;
+    }
+    else
+    {
+        return false;
+    }
+}
+```
+
+## Create and setup mocks
+To start, create a mock for the `FormRecognizerClient`. Most methods in this service make use of [Long-Running Operations][lros], so you'll likely need to create a mock operation as well.
+
+```C# Snippet:FormRecognizerCreateMocks
+var fakeModelId = Guid.NewGuid().ToString();
+var fakeOperationId = $"{fakeModelId}/analyzeResults/{Guid.NewGuid()}";
+var mockClient = new Mock<FormRecognizerClient>();
+var mockOperation = new Mock<RecognizeCustomFormsOperation>(fakeOperationId, mockClient.Object);
```

Comment:
I really don't like that we need to create a `fakeOperationId` to mock the operation object. Also, if we wanted to mock only the operation (without a previous `StartRecognizeCustomForms` call), we would still need to create a mock client to instantiate the operation ([related discussion](https://github.com/Azure/azure-sdk/pull/2121#discussion_r537899459)).

Should we wait on https://github.com/Azure/azure-sdk/pull/2121 before adding mockable constructors to our LROs?

Reply:
Yeah you are right. Makes it inconvenient. 
I think it is better to add it now in case a customer needs it, and ping that PR to get a conclusion.
Whenever they reach it, the sample can be updated

Reply:
Added protected constructors! Pinged Pavel in the guidelines PR.